### PR TITLE
⬆️ Upgrade babel & jest

### PR DESCRIPTION
**Pull request by dmnd**
It turns out when I ran prettier on everything I broke a test because
the version of babel in use doesn't support Flow's syntax for specifying
generic parameters when calling a function.

So I upgraded everything and now tests are green again.

@zgotsch @FermiDirak can you think of anything else I shoud test?

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1,3 +1,35 @@
...
+  ],
+  "plugins": [
+    "@babel/plugin-syntax-dynamic-import",
+    "@babel/plugin-syntax-import-meta",
+    "@babel/plugin-proposal-class-properties",
+    "@babel/plugin-proposal-json-strings",
+    [
+      "@babel/plugin-proposal-decorators",
+      {
+        "legacy": true
```
Comment (FermiDirak) [inline .babelrc]: is this config necessary?

File: .babelrc [hunk 2]
```diff
@@ -1,3 +1,35 @@
 {
-  "presets": ["env", "stage-0", "react", "flow"]
```
Comment (FermiDirak) [inline .babelrc]: What's the rational behind not keeping `preset-stage-0`?

Comment (dmnd) [reply .babelrc]: The `state-0` preset is deprecated. `babel-upgrade` replaced it by inlining the set of plugins. Probably not all of them are used (yet?), but I figured this doesn't matter.

Comment (FermiDirak) [reply .babelrc]: That makes sense 👍 LGTM

**Pull request merged**